inc/classes/class-manifest.php

```diff
@@ -41,9 +41,11 @@ public function filter_web_app_manifest( $manifest ) {
 		$icon_sizes = array( '72', '96', '128', '144', '152', '192', '384', '512' );
 
 		foreach ( $icon_sizes as $icon_size ) {
+			$img_url     = sprintf( '%1$s/assets/img/icon-%2$sx%2$s.png', get_template_directory_uri(), $icon_size );
+			$filter_name = sprintf( 'rt_pwa_extensions_app_icon_%1$s_%1$s', $icon_size );
 
 			$manifest['icons'][] = array(
-				'src'   => sprintf( '%1$s/assets/img/icon-%2$sx%2$s.png', get_template_directory_uri(), $icon_size ),
+				'src'   => apply_filters( $filter_name, $img_url ),
```

Comment:
@deepaklalwani97 @pradeep910 Why do we need separate filter for icons? Anyone can use `web_app_manifest` filter from PWA plugin to change icons same as we did here.

Reply:
It's for exact size URL change. filter will allow users to change specific size URL as there can be device support issues. Like iOS, etc.

Reply:
@pradeep910 That also can be done using `web_app_manifest` filter. But I don't see any issue by adding new filter in our plugin. It was suggestion.